airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/avro/JsonToAvroSchemaConverter.java

```diff
@@ -240,6 +254,11 @@ Schema getNullableFieldTypes(final String fieldName, final JsonNode fieldDefinit
       if (!nonNullFieldTypes.contains(NULL_SCHEMA)) {
         nonNullFieldTypes.add(0, NULL_SCHEMA);
       }
+      if ((nonNullFieldTypes
+          .stream().anyMatch(schema -> schema.getLogicalType() != null)) &&
+          (!nonNullFieldTypes.contains(Schema.create(Schema.Type.STRING)))) {
+        nonNullFieldTypes.addLast(Schema.create(Schema.Type.STRING));
```

Comment:
Changing the `nonNullFieldTypes` from the generic `List` to `LinkedList` does not seem necessary. The `add` method on the `List` interface is equivalent to `addLast`.

Reply:
done